src/App.js

```diff
@@ -0,0 +1,31 @@
+import React, { useState, useEffect } from 'react';
+import AddTodoForm from './AddTodoForm';
+import TodoList from './TodoList';
+
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem("savedTodoList")) || []);
+
+  useEffect(() => {
+    localStorage.setItem("savedTodoList", JSON.stringify(todoList))
+  }, [todoList] )
+
+  return [todoList, setTodoList]
```

Comment:
nit: be consistent on semicolon use. Tools such as eslint and Prettier can assist with this.